ttnn/cpp/ttnn/operations/uniform/device/kernels/writer_uniform.cpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+
+#include "common/constants.hpp"
+#include "dataflow_api.h"
+
+using namespace tt;
+
+void kernel_main() {
+    constexpr uint32_t intermed_cb_id = get_compile_time_arg_val(0);
+    constexpr uint32_t dst_cb_id = get_compile_time_arg_val(1);
+    constexpr bool output_is_dram = get_compile_time_arg_val(2) == 1;
+
+    uint32_t dst_addr = get_arg_val<uint32_t>(0);
+    union {
+        float f;
+        uint32_t u;
+    } f2u_from, f2u_to;
+    f2u_from.u = get_arg_val<uint32_t>(1);
+    f2u_to.u = get_arg_val<uint32_t>(2);
+    uint32_t start_id = get_arg_val<uint32_t>(3);
+    uint32_t num_tiles = get_arg_val<uint32_t>(4);
+    uint32_t end_id = start_id + num_tiles;
+
+    const InterleavedAddrGenFast<output_is_dram> output_addrg = {
```

Comment:
I think this could be done on the SFPU and it would be much faster.  What if we generate a random number on SFPU then shift it 9 bits to the right, so that  we get the mantissa part of a fp32 number, then in the exponent section we put the appropriate exponent such that our number is between 1 and 2, and then we subtract 1 from it using a programmable constant register. Then we have a value between 0 and 1 . 

Then we use eltwise operations to multiply with the range and then add to the from constant. This may require careful consideration of the bounds, but this will be much faster as things will be done in parallel and without trips to L1

Reply:
Thank for your suggestion, I've updated the sfpu API to generate random float instead of uint32.